### PR TITLE
chore(ec2): dl2q instance type, `IntanceType` fixes

### PR DESCRIPTION
**Pull request by nmussy**
### Issue # (if applicable)

None

### Reason for this change

Follow up to #29427. Added a missing instance type, and minor fixes to the `IntanceType` class

### Description of changes

I'd missed an instance type, [`dl2q`](https://aws.amazon.com/ec2/instance-types/dl2q/), which was neither in `us-east-1` or `us-east-2`, but in `us-west-2`.

I've also added a couple of missing symbolic names, as well as fixed some differing comments between the key and its symbolic value (e.g. `M3` and `STANDARD3`)

I also re-ordered a couple of enum values, when the symbolic value was separated from its match

### Description of how you validated changes

Compared the CDK to the SDK to find the missing instance. Programmatically iterated over the comments of `IntanceType` to make sure the comments of the symbolic keys matched the one below. 

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts [hunk 1]
```diff
@@ -611,20 +611,30 @@ export enum InstanceClass {
...
+  /**
+   * Storage/compute balanced instances, 1st generation
+   */
+  H1 = 'h1',
+
   /**
    * High performance computing powered by AWS Trainium
    */
-  TRN1 = 'trn1',
+  TRAINING_ACCELERATOR1 = 'training-accelerator1',
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:622]: Where are these descriptions pulled from? Some of them seem mismatched after the changes, or were they incorrect before?

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts [hunk 2]
```diff
@@ -611,20 +611,30 @@ export enum InstanceClass {
...
-   * Storage/compute balanced instances, 1st generation
+   * Network-optimized high performance computing powered by AWS Trainium
    */
-  H1 = 'h1',
+  TRAINING_ACCELERATOR1_ENHANCED_NETWORK = 'training-accelerator1-enhanced-network',
+
+  /**
+   * Network-optimized high performance computing powered by AWS Trainium
+   */
+  TRN1N = 'trn1n',
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:637]: Description for this has changed.

**Review by paulhcsun — COMMENTED** [review]
Just a couple questions.

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts (hunk 1 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:622]: Neither `trn1` nor `trn1n` had a symbolic "alias" given to them when they were added, unlike every other instance class. I added one, didn't get it from anywhere really, looked up the AWS docs and tried to match the other's.

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts (hunk 2 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:637]: Yeah, the `trn1n` description didn't have the network optimization discriminator, both `trn1*` had the same description: https://github.com/aws/aws-cdk/blob/fac4a9c23f8e9090b3dc7e26a8306d3a8034b4c9/packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts#L614-L622

Comment (paulhcsun) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:637]: Gotcha, thanks for clarifying

**Review by paulhcsun — DISMISSED** [review]

**Review by paulhcsun — APPROVED** [review]

**Pull request merged**